jsonrpc/provider/src/provider.ts

```diff
@@ -109,13 +112,15 @@ export class JsonRpcProvider extends IJsonRpcProvider {
     }
     this.connection = this.setConnection(connection);
     await this.connection.open();
-    this.registerEventListeners();
+    if (!this.hasRegisteredEventListeners) {
+      this.registerEventListeners();
+      this.hasRegisteredEventListeners = true;
+    }
     this.events.emit("connect");
   }
 
   protected async close() {
     await this.connection.close();
-    this.events.emit("disconnect");
```

Comment:
I've removed this explicit `emit` because it's duplicating the `disconnect` event that will automatically be emitted by the listener already set for `this.connection.close` below in `registerEventListeners`: 

```
this.connection.on("close", () => this.events.emit("disconnect"));
```